app/indexers/collection_indexer.rb

```diff
@@ -3,7 +3,11 @@ class CollectionIndexer
   delegate :query_service, to: :metadata_adapter
   attr_reader :resource
   def initialize(resource:)
-    @resource = resource
+    @resource = if resource.is_a?(DeletionMarker)
```

Comment:
I wonder if we should delegate `member_of_collection_ids` on DeletionMarker so we don't have to branch on class?